shutdown.go

```diff
@@ -0,0 +1,65 @@
+package gokit
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// Server will create a http.Server from the Go standard library
+type Server struct {
+	Addr    string        // TCP address to listen on, ":http" if empty
+	Handler http.Handler  // handler to invoke, http.DefaultServeMux if nil
+	Timeout time.Duration // Timeout to wait for shutdown
+}
+
+// ListenAndServe mirrors the function from the Go standard library
+func (s *Server) ListenAndServe() string {
```

Comment:
The Go Standard library returns `error`, no?

Reply:
Yes, it does. And usually we fix it around one of our loggers, like so `log.Println(s.ListenAndServe())`. So it makes no difference to the logger if we return a string. I thought that since the return value is used for logging and not actually to return an error because they're handled in the function, returning a string will be more succinct. What do you think?

Reply:
I think we should let the caller decide what it wants to do with the error.

Reply:
After discussion, we decided to change the signature to be exactly the same as `http.Server` from the standard library for this to be a drop-in replacement